api/python/provisioner/commands/setup_provisioner.py

```diff
@@ -1299,39 +1299,49 @@ def _run(self, nodes, **kwargs):  # noqa: C901 FIXME
             ), targets=run_args.primary.minion_id
         )
 
-        if run_args.target_build:
-            logger.info("Updating target build pillar")
-            ssh_client.cmd_run(
-                (
-                    'provisioner pillar_set --fpath release.sls'
-                    f' release/target_build \'"{run_args.target_build}"\''
-                ), targets=run_args.primary.minion_id
-            )
-
-            logger.info("Get release factory version")
+        logger.info("Updating target build pillar")
+        if run_args.url_cortx:
+            url = run_args.url_cortx
+        elif run_args.target_build:
             if run_args.dist_type == config.DistrType.BUNDLE:
                 url = f"{run_args.target_build}/cortx_iso"
             else:
                 url = run_args.target_build
+        elif run_args.iso_cortx:  # Fix Me update with proper path
+            logger.info("Update target with mounted path")
```

Comment:
In that case url would be undefined and we will fail few lines later. Please test that code.